Automattic/wp-calypso
# Theme Showcase: Visiting the showcase while logged in forces you to select a site.

#### Changes proposed in this Pull Request

* Visiting /themes while logged in forces you to select a site

#### Testing instructions

* Visit `/themes` while logged out
  *  Should still see the themes showcase
* Visit `/themes` while logged out and javascript is off
  *  Should still see the themes showcase (Server side rendered)
* Visit `/themes` while logged in
  * You should be forced to choose a site to proceed, and end up on a url like `/themes/<siteurl>`.

#### Dev Issues

Note: I am seeing some weirdness on the site selection page. However, I believe it is an artifact of isometric rendered pages (server side + client side) in the current dev environment.  Screenshot:

![2021-05-07_16-03](https://user-images.githubusercontent.com/937354/117508241-df61c000-af4d-11eb-8292-409badc08c89.png)

I don't believe it is related to this branch, because I see the same thing on the current trunk commit with no changes:

![2021-05-07_13-58](https://user-images.githubusercontent.com/937354/117508272-ebe61880-af4d-11eb-95b5-24c46317a5f1.png)

Notice production renders fine, but in dev, the same page under the same conditions does not render properly. I believe there is some sort of bug when the client side render 'takes over' the server side render.  

In short, without any changes in this PR, I am seeing `/themes` render quite differently in prod vs. localhost dev while logged in. Anyone know any more about this? (`server-side-rendering.md` mentioned these gurus: @ehg @mcsf @ockham @seear)

Edit: Reasoning for this discrepancy is explained here: pd2qbF-P-p2

**To temporarily fix the issue for localhost testing, also apply this patch to your local installation**:
```diff
diff --git a/client/controller/web-util.js b/client/controller/web-util.js
index 55d62fe50e..20940e794b 100644
--- a/client/controller/web-util.js
+++ b/client/controller/web-util.js
@@ -8,5 +8,5 @@ export function render( context ) {
 }
 
 export function hydrate( context ) {
-       ReactDom.hydrate( context.layout, document.getElementById( 'wpcom' ) );
+       ReactDom.render( context.layout, document.getElementById( 'wpcom' ) );
 }
```
A larger fix for this is in the works in #52731

Related to #14896
